src/test/java/org/broadinstitute/consent/http/service/DaaServiceTest.java

```diff
@@ -103,6 +117,91 @@ void testRemoveDacFromDaa() {
     service.removeDacFromDaa(1, 1);
   }
 
+  @Test
+  void testSendDaaRequestEmailsNoSigningOfficials() throws Exception {
+    User user = mock(User.class);
+    when(user.getInstitutionId()).thenReturn(1);
+
+    Institution institution = mock(Institution.class);
+    when(institution.getId()).thenReturn(1);
+    when(institution.getSigningOfficials()).thenReturn(List.of());
+
+    when(institutionService.findAllInstitutions()).thenReturn(
+        Collections.singletonList(institution));
+
+    initService();
+
+    DaaService daaSpy = spy(service);
+    assertThrows(NotFoundException.class, () -> daaSpy.sendDaaRequestEmails(user, 1));
+    verify(daaSpy, times(1)).sendDaaRequestEmails(any(), any());
```

Comment:
When I find myself using Spys in tests, my spidey senses tingle 😄 they often point to a need to reorganize my code, or they are not needed.

In this case and throughout these new tests, I don't think you need to Spy the `service`: you are explicitly calling the method that you are verifying, so I don't think the verification gets you anything extra.

Reply:
In service tests, there are often conditionals that are embedded in the service logic such that a method call may or may not be made depending on input. In those kinds of situations, I find `spy`s useful to ensure that we ARE calling a method in one case and NOT calling it in a different case. In this situation, I agree with @okotsopoulos in that we're probably over testing this particular path.